lib/stdlib/src/lists.erl

```diff
@@ -391,6 +392,35 @@ delete(_, []) -> [].
 zip([X | Xs], [Y | Ys]) -> [{X, Y} | zip(Xs, Ys)];
 zip([], []) -> [].
 
+%% Return [{A0, ..., Z0}, ..., {An, ..., Zn}] for lists [A0, ..., An],
+%% [B0, ..., Bn], ..., [Z0, ..., Zn].
+
+-spec transpose(Lists) -> Lists when
+      Lists :: [[A]],
+      A :: term().
+
+transpose([]) -> [];
+transpose([List|_] = Lists) ->
+    ListLen = length(List),
+    case all(fun(ListN) -> ListLen =:= length(ListN) end, Lists) of
```

Comment:
I think there is no need to calculte and check the length of the 1st list twice?

```suggestion
transpose([List|Tail] = Lists) ->
    ListLen = length(List),
    case all(fun(ListN) -> ListLen =:= length(ListN) end, Tail) of
```

Also, can this check be skipped entirely and error detected at some later stage? `length(ListN)` is O(n), so, might be noticeable on long lists.

Reply:
Added. Thanks @seriyps